### ISSUE TITLE
OverlayPanel:  dismissable attribute not working in 6.3

### ISSUE THREAD

**Issue by erickdeoliveiraleal** [primefaces/primefaces#3861]
## 1) Environment
- PrimeFaces version: 6.3 SNAPSHOT
- Does it work on the newest released PrimeFaces version? Version? yes
- Does it work on the newest sources in GitHub? no
- Application server + version: wildfly
- Affected browsers: all

## 2) Expected behavior

clicking outside overlaypanel  should close it

## 3) Actual behavior

overlayPanel remains opened

## 4) Steps to reproduce

click on button, click outside button

## 5) Sample XHTML

```
<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml"
	xmlns:ui="http://java.sun.com/jsf/facelets"
	xmlns:f="http://java.sun.com/jsf/core"
	xmlns:p="http://primefaces.org/ui"
	xmlns:h="http://java.sun.com/jsf/html">

<h:head>
	<title>PrimeFaces Test</title>
</h:head>
<h:body>

	<p:commandButton title="Setores" icon="fa fa-ellipsis-h"
		oncomplete="PF('setores').show('#{component.clientId}')" update="setores" />


	<p:overlayPanel widgetVar="setores" id="setores" dismissable="true">


		<h1>Sorry, I'm bug</h1>


	</p:overlayPanel>

</h:body>
</html>

```

## 6) Sample bean
not needed

## 7) Test
https://github.com/erickdeoliveiraleal/primefaces-test/tree/dismissable

**Comment by tandraschko** [comment]
works fine.

**Comment by tandraschko** [comment]
setting it to true, hides the panel when clicking outside. setting it to false doesn't close it.

**Comment by erickdeoliveiraleal** [comment]
Sorry. I added a new code, can you test it? I can see the bug now. Only occurs with update=something.

**Comment by tandraschko** [comment]
Next time pls test it before you create the issue.

**Comment by tandraschko** [comment]
works just fine.

**Comment by erickdeoliveiraleal** [comment]
@tandraschko please, reopen this issue, I included a test https://github.com/erickdeoliveiraleal/primefaces-test/tree/dismissable
it was missing update attribute.

**Comment by erickdeoliveiraleal** [comment]
@tandraschko please, reopen this issue, I included a test https://github.com/erickdeoliveiraleal/primefaces-test/tree/dismissable
it was missing update attribute.

**Comment by erickdeoliveiraleal** [comment]
@tandraschko please, reopen this issue, I included a test https://github.com/erickdeoliveiraleal/primefaces-test/tree/dismissable
it was missing update attribute.

**Comment by tandraschko** [comment]
it just works for me.
When i click outside the overlay panel, it closes.

Please, don't waste my time - I already tested your code 3 times now and everytime it just works!
Try to build again on localhost, clear caches etc. I can't reproduce it.

**Comment by erickdeoliveiraleal** [comment]
@tandraschko i retested it many times, different machines, this bug exists.
You can test it from here: https://github.com/erickdeoliveiraleal/primefaces-test/tree/dismissable

**Comment by tandraschko** [comment]
For me the bug exists sometimes but it's random.

AFAICS it could be a bug in jQuery and i have no idea how to solve it.

please go into core.utils.js and search for "registerHideOverlayHandler"

my test code now looks like:

`        registerHideOverlayHandler: function(widget, hideNamespace, overlay, resolveIgnoredElementsCallback, hideCallback) {

            console.log('registerHideOverlayHandler: ' + hideNamespace);

            widget.addDestroyListener(function() {
                $(document).off(hideNamespace);
            });

            $(document).off(hideNamespace).on(hideNamespace, function (e) {
                console.log('hide handler');
......`

**Comment by tandraschko** [comment]
after this changes you will see the following:
- on page load it prints "registerHideOverlayHandler..."
- every click on the document prints "hide handler" on the console

after the ajax update:
- it prints again: "registerHideOverlayHandler...."
- but the "hide handler" is never printed again on any click, which means that the off() call works fine but on() not.

So somehow jquery doesn't execute the listeners on the document again.
Please investigate and let me know if you can find something.

**Comment by tandraschko** [comment]
@melloware @jxmai @Rapster anyone got a idea or would like to play a bit, too?

**Comment by Rapster** [comment]
Sorry I don't have time for this :(

**Comment by tandraschko** [comment]
If you use the inspector on the events on the document, you can see that it mousedown is not added again.
Maybe a stupid jquery bug...

**Comment by tandraschko** [comment]
found the reason, nvm :D

**Comment by tandraschko** [comment]
@erickdeoliveiraleal you own me a beer someday, invested almost 3 hours for this bug.

**Comment by melloware** [comment]
Great work!

**Comment by erickdeoliveiraleal** [comment]
you deserve many beers, my friend. thanks!

**Comment by tandraschko** [comment]
I will ping you when i'm brasil the next time :D

**Comment by jxmai** [comment]
Just checked github  :) sorry for late reply. Great work Thomas ! :+1:

**Comment by erickdeoliveiraleal** [comment]
fixed in #4122
